content/post/2025-06-19-open-files-folders-websites/index.Rmd

```diff
@@ -0,0 +1,154 @@
+
+```{r, eval=FALSE}
+cli::cli_alert_success("Hey go edit {.file config.toml} please!")
+```
+> _"If the terminal supports ANSI hyperlinks (e.g. RStudio, iTerm2, etc.), then cli creates a clickable link that opens the file in RStudio or with the default app for the file type."_
+
+[^adventurous]: Look how adventurous I was when preparing this post! Running R from a terminal!
+[^atom]: Which is... [Atom](https://en.wikipedia.org/wiki/Atom_(text_editor)), meaning I live in the past since its end-of-life occurred a few years ago. :ghost: I actually rarely use it.
+
+## Open the script or test file of a script-test pair: `usethis::use_r()` and `usethis::use_test()`
+
+Coming back to the wonderful usethis package, when you name your test files after your scripts as you should,
+running [`usethis::use_test()`](https://r-pkgs.org/testing-basics.html#create-a-test) when focussed on an R script will open its test file, and conversely `usethis::use_r()` when focussed on a test file.
+This is still how I switch between the two even if inside Positron.
+
+## Open a project: `positron`, project launcher, {usethis}
+
+This is all very good when within a project, but how do you enter your IDE in the first place?
+How sad would it be to lose momentum before even launching a project?
+
+When I used RStudio IDE I would navigate to the folder of interest then double-click on the `.Rproj` file.
+Hannah Frick inspired me, when demoing Positron to me, to think a bit harder about how much I click. :sweat_smile:
+I have adopted a more keyboard-based workflow since I switched to Positron, which I could have done with RStudio IDE.
+
+One thing in particular that Hannah showed me that looks cool is the use of a [project launcher](https://positron.posit.co/rstudio-rproj-file.html#use-an-application-launcher).
+
+Now, I myself still use the terminal. 
+For instance if I clone a repository[^usethis], I'll then open it with the `positron` command.
+
+Another way to open projects I use a lot is, on Positron as on Rstudio, clicking to the recent projects -- when the IDE is already up and running.
+Unfortunately, I created and [currently](https://stateofther.netlify.app/post/saperlipopette/) [often](https://www.meetup.com/rbuenosaires/events/308338205/) use a [package that creates throwaway folders for practicing Git](https://docs.ropensci.org/saperlipopette/) which makes the list of recent projects utter rubbish. 🙃
+
+Within saperlipopette itself, to create and open the exercice folder on behalf of the user I use `usethis::create_project()`, which handily opens it in Positron when run in Positron and RStudio when run in RStudio.
+
+[^usethis]: I know I could use `usethis::create_from_github()` and sometimes do but not always... Maybe the message of this post is that my workflows are a big mess.
+
+## Open an URL in the browser: `browseURL()` and {cli}
```

Comment:
The `{cli}` is less surprising than the first time round, but I would adapt this one in the same style if you change the first.